app/Http/Controllers/ClientController.php

```diff
@@ -1,28 +1,28 @@
 <?php
 
+declare(strict_types=1);
+
 namespace App\Http\Controllers;
 
 use App\Models\Product;
 use App\Models\Service;
 use App\Repositories\Interfaces\ProductRepositoryInterface;
+use Illuminate\View\View;
 
 class ClientController extends Controller
 {
-    private ProductRepositoryInterface $productRepository;
-
-    public function __construct(ProductRepositoryInterface $productRepository)
-    {
-        $this->productRepository = $productRepository;
-    }
+    public function __construct(
+        protected readonly ProductRepositoryInterface $productRepository
+    ) {}
 
-    public function index()
+    public function index(): View
     {
         $products = $this->productRepository->all();
 
         return view('client.index', compact('products'));
```

Comment:
```
return view('client.index', [
    'products' => $products,
]);
```

Reply:
return view('client.index', [
    'products' => $this->productRepository->all(),
]);

Reply:
fixed